reporters/kamon-datadog/src/main/scala/kamon/datadog/package.scala

```diff
@@ -97,12 +98,14 @@ package object datadog {
 
     // Apparently okhttp doesn't require explicit closing of the connection
     private def createHttpClient(): OkHttpClient = {
-      new OkHttpClient.Builder()
+      val builder = new OkHttpClient.Builder()
         .connectTimeout(connectTimeout.toMillis, TimeUnit.MILLISECONDS)
         .readTimeout(readTimeout.toMillis, TimeUnit.MILLISECONDS)
         .writeTimeout(writeTimeout.toMillis, TimeUnit.MILLISECONDS)
         .retryOnConnectionFailure(false)
-        .build()
+
+      if (usingCompression) builder.addInterceptor(new DeflateInterceptor).build()
+      else builder.build()
```

Comment:
It's a mutable API so in theory you could do this:

```suggestion
      if (usingCompression)
        builder.addInterceptor(new DeflateInterceptor)
      
      builder.build()
```

But your version is more idiomatic I think.